core/src/main/java/org/infinispan/cache/impl/TypeConverterDelegatingAdvancedCache.java

```diff
@@ -744,4 +744,12 @@ public void replaceAll(BiFunction<? super K, ? super V, ? extends V> function) {
       return returned;
    }
 
+   @Override
+   public AdvancedCache<K, V> lockAs(Object lockOwner) {
+      AdvancedCache<K, V> returned = super.lockAs(lockOwner);
+      if (returned != this && returned instanceof TypeConverterDelegatingAdvancedCache) {
+         ((TypeConverterDelegatingAdvancedCache) returned).entryFactory = this.entryFactory;
```

Comment:
It can't be as the wrapper function is only given the wrapped `AdvancedCache`. It doesn't have access to the old wrapper and thus the factory needed.